src/hotspot/cpu/s390/sharedRuntime_s390.cpp

```diff
@@ -2692,7 +2690,8 @@ void SharedRuntime::generate_deopt_blob() {
   __ z_std(Z_FRET, offset_of(frame::z_abi_160_spill, spill[1]), Z_SP);
 
   // let the unpacker layout information in the skeletal frames just allocated.
-  __ get_PC(Z_RET);
+  offs = __ offset();
```

Comment:
Shouldn't we use the offset from above instead of the current one? (PPC64 does that.) I think this line should get removed.

Reply:
I have tried that using same offset from above instead current one but it's not  pointing to correct PC for oop Maps.

Reply:
Setting last_Java_pc = pc() + current offset looks weird. Could it be that this breaks something which prevents the failure from happening? E.g. pointing to somewhere outside the deopt blob?

Reply:
I think you can't use this get_PC version for the intended purpose. It doesn't set Z_RET = start pc + offs which is what you probably wanted.

Reply:
if Z_RET = start pc + offs can not be used Let me explore on how to get that offset on s390x.

Reply:
I think something like `__ get_PC(Z_R1_scratch, oop_map_offs - __ offset());` should work with oop_map_offs taken from above where the `add_gc_map` is used.

Reply:
@TheRealMDoerr I have tried your suggestion and it' s works. I am not seeing any regressions. Added a commit for this change.